### PR TITLE
U4X-418: Add ability to display the factor of the packaging unit on the stock items and their transactions 

### PR DESCRIPTION
**Pull request by slubwama**
https://metsprogramme.atlassian.net/browse/U4X-418

### CODE REVIEW

File: owa/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "stockmanagement-app",
   "version": "0.1.0",
   "private": true,
-  "proxy": "http://127.0.0.1:8080/",
+  "proxy": "http://127.0.0.1:8084/",
```
Comment (Mwanje) [inline owa/package.json:5]: @slubwama Could stick with the default port

File: owa/src/features/stockItems/StockItemBatchInformation.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ const StockItemBatchInformationTable: React.FC<StockItemBatchInformationTablePro
       location: row?.partyName,
       quantity: row?.quantity?.toLocaleString() ?? "",
       batchno: row.batchNumber ?? "",
-      quantityuom: row.quantityUoM ?? ""
+      quantityuom: `${row.quantityUoM ?? ""} of  ${row.quantityFactor ?? ""}`
```
Comment (akileng56) [inline owa/src/features/stockItems/StockItemBatchInformation.tsx:64]: ``${row.quantityUoM ?? ""} ${row?.quantityFactor ? of ${row?.quantityFactor} : ""}``

File: owa/src/features/stockItems/EditStockItem.tsx [hunk 3]
```diff
@@ -645,7 +645,7 @@ export const EditStockItem: React.FC<EditStockItemProps> = ({
                                 id="purchasePriceUoMUuid" value={model.purchasePriceUoMUuid ?? "placeholder-item"} onChange={onPurchasePriceUoMChange}>
                                 <SelectItem value="placeholder-item" text={t("stockmanagement.notset")} />
                                 {packagingUnits?.map(uom => {
-                                    return <SelectItem key={uom.uuid} value={uom.uuid} text={uom.packagingUomName ?? ""} />
+                                    return <SelectItem key={uom.uuid} value={uom.uuid} text={`${uom.packagingUomName ?? ""} - ${uom.factor ?? ""}`} />
```
Comment (akileng56) [inline owa/src/features/stockItems/EditStockItem.tsx:648]: Same here

File: owa/src/features/stockItems/StockItemQuantities.tsx [hunk 4]
```diff
@@ -57,7 +57,7 @@ const StockItemQuantitiesTable: React.FC<StockItemQuantitiesTableProps> = ({
       uuid: `${row.partyUuid}${row.stockBatchUuid}${index}`,
       location: row?.partyName,
       quantity: row?.quantity?.toLocaleString() ?? "",
-      quantityuom: row.quantityUoM ?? ""
+      quantityuom: `${row.quantityUoM ?? ""} of  ${row.quantityFactor ?? ""}`
```
Comment (akileng56) [inline owa/src/features/stockItems/StockItemQuantities.tsx:60]: Same here

File: owa/src/features/stockItems/StockItemQuantities.tsx [hunk 5]
```diff
@@ -106,7 +106,7 @@ const StockItemQuantitiesTable: React.FC<StockItemQuantitiesTableProps> = ({
                 <TableRow className='table-totals'>
                   <TableCell>Total</TableCell>
                   <TableCell>{total?.quantity?.toLocaleString() ?? ""}</TableCell>
-                  <TableCell>{total?.quantityUoM ?? ""}</TableCell>
+                  <TableCell>{total?.quantityUoM ?? ""} of {total?.quantityFactor ?? ""}</TableCell>
```
Comment (akileng56) [inline owa/src/features/stockItems/StockItemQuantities.tsx:109]: Same here

File: owa/src/features/stockOperations/StockOperationItemsTable.tsx [hunk 6]
```diff
@@ -648,11 +648,11 @@ const StockOperationItemsTable: React.FC<StockOperationItemTableProps> = ({
                                                     onToggleClick={(e => handleStockItemPackagingUoMsSearch(row, ""))}
                                                     shouldFilterItem={(data) => true}
                                                     onInputChange={(q) => handleStockItemPackagingUoMsSearch(row, q)}
-                                                    itemToString={item => (item?.display ?? item?.packagingUomName ?? '')}
+                                                    itemToString={item => (item?.display ?? item?.packagingUomName ?? '')+"-"+(item?.factor ?? '')}
                                                     placeholder={'Filter...'}
                                                     invalid={(row.uuid in errors) && ("stockItemPackagingUOMUuid" in errors[row.uuid]) && !errors[row.uuid]["stockItemPackagingUOMUuid"]}
                                                 />}
-                                                {!canEdit && row?.stockItemPackagingUOMName}
+                                                {!canEdit && (row?.stockItemPackagingUOMName)+"-"+(row?.stockItemPackagingUOMFactor)}
```
Comment (akileng56) [inline owa/src/features/stockOperations/StockOperationItemsTable.tsx:655]: Same here

File: owa/src/features/stockOperations/StockOperationItemsTable.tsx [hunk 7]
```diff
@@ -648,11 +648,11 @@ const StockOperationItemsTable: React.FC<StockOperationItemTableProps> = ({
                                                     onToggleClick={(e => handleStockItemPackagingUoMsSearch(row, ""))}
                                                     shouldFilterItem={(data) => true}
                                                     onInputChange={(q) => handleStockItemPackagingUoMsSearch(row, q)}
-                                                    itemToString={item => (item?.display ?? item?.packagingUomName ?? '')}
+                                                    itemToString={item => (item?.display ?? item?.packagingUomName ?? '')+"-"+(item?.factor ?? '')}
```
Comment (akileng56) [inline owa/src/features/stockOperations/StockOperationItemsTable.tsx:651]: Same here

File: owa/src/features/stockOperations/StockOperationItemsTable.tsx [hunk 8]
```diff
@@ -638,7 +638,7 @@ const StockOperationItemsTable: React.FC<StockOperationItemTableProps> = ({
                                                 {canEdit && <ComboBox titleText="" size='sm' id={`quantityuom-${row.uuid}`}
                                                     initialSelectedItem={(row?.stockItemPackagingUOMUuid) ? {
                                                         uuid: row?.stockItemPackagingUOMUuid,
-                                                        display: row?.stockItemPackagingUOMName
+                                                        display: (row?.stockItemPackagingUOMName)+"-"+(row?.stockItemPackagingUOMFactor)
```
Comment (akileng56) [inline owa/src/features/stockOperations/StockOperationItemsTable.tsx:641]: Same here

File: owa/src/features/stockItems/StockItemTransactions.tsx [hunk 9]
```diff
@@ -90,8 +90,8 @@ const StockItemTransactionsTable: React.FC<StockItemTransactionsTableProps> = ({
...
       reference: stockItemTransaction?.stockOperationUuid ? <Link to={URL_STOCK_OPERATION(stockItemTransaction?.stockOperationUuid!)} target={"_blank"}>{`${stockItemTransaction?.stockOperationNumber}`}</Link> : <></>,
       status: stockItemTransaction?.stockOperationStatus ?? "",
-      qtyin: stockItemTransaction?.quantity >= 0 ? `${stockItemTransaction?.quantity?.toLocaleString()} ${stockItemTransaction?.packagingUomName ?? ""}` : "",
-      qtyout: stockItemTransaction?.quantity < 0 ? `${(-1 * stockItemTransaction?.quantity)?.toLocaleString()} ${stockItemTransaction?.packagingUomName ?? ""}` : ""
+      qtyin: stockItemTransaction?.quantity >= 0 ? `${stockItemTransaction?.quantity?.toLocaleString()} ${stockItemTransaction?.packagingUomName ?? ""} of ${stockItemTransaction?.packagingUomFactor ?? ""}` : "",
+      qtyout: stockItemTransaction?.quantity < 0 ? `${(-1 * stockItemTransaction?.quantity)?.toLocaleString()} ${stockItemTransaction?.packagingUomName ?? ""} of ${stockItemTransaction?.packagingUomFactor ?? ""}` : ""
```
Comment (akileng56) [inline owa/src/features/stockItems/StockItemTransactions.tsx:94]: Same here

File: owa/src/features/stockItems/StockItemTransactions.tsx [hunk 10]
```diff
@@ -90,8 +90,8 @@ const StockItemTransactionsTable: React.FC<StockItemTransactionsTableProps> = ({
       batchno: stockItemTransaction.stockBatchNo ? `${stockItemTransaction.stockBatchNo}${stockItemTransaction.expiration ? ` (${formatDisplayDate(stockItemTransaction.expiration)})` : ""}` : "",
       reference: stockItemTransaction?.stockOperationUuid ? <Link to={URL_STOCK_OPERATION(stockItemTransaction?.stockOperationUuid!)} target={"_blank"}>{`${stockItemTransaction?.stockOperationNumber}`}</Link> : <></>,
       status: stockItemTransaction?.stockOperationStatus ?? "",
-      qtyin: stockItemTransaction?.quantity >= 0 ? `${stockItemTransaction?.quantity?.toLocaleString()} ${stockItemTransaction?.packagingUomName ?? ""}` : "",
-      qtyout: stockItemTransaction?.quantity < 0 ? `${(-1 * stockItemTransaction?.quantity)?.toLocaleString()} ${stockItemTransaction?.packagingUomName ?? ""}` : ""
+      qtyin: stockItemTransaction?.quantity >= 0 ? `${stockItemTransaction?.quantity?.toLocaleString()} ${stockItemTransaction?.packagingUomName ?? ""} of ${stockItemTransaction?.packagingUomFactor ?? ""}` : "",
```
Comment (akileng56) [inline owa/src/features/stockItems/StockItemTransactions.tsx:93]: Same here

File: owa/src/features/stockItems/StockItemTable.tsx [hunk 11]
```diff
@@ -146,9 +146,9 @@ const StockItemTable: React.FC<StockItemTableProps> = ({
       retired: stockItem?.voided ? t("stockmanagement.yes") : t("stockmanagement.no"),
       isdrug: stockItem?.drugUuid ? t("stockmanagement.yes") : t("stockmanagement.no"),
       preferredVendorName: stockItem?.preferredVendorName,
-      defaultStockOperationsUoMName: stockItem?.defaultStockOperationsUoMName,
+      defaultStockOperationsUoMName: `${stockItem?.defaultStockOperationsUoMName} of ${stockItem?.defaultStockOperationsUoMFactor}`,
       dispensingUnitName: stockItem?.dispensingUnitName,
-      reorderLevel: stockItem?.reorderLevelUoMName && stockItem?.reorderLevel ? `${stockItem?.reorderLevel?.toLocaleString()} ${stockItem?.reorderLevelUoMName}` : "",
+      reorderLevel: stockItem?.reorderLevelUoMName && stockItem?.reorderLevel ? `${stockItem?.reorderLevel?.toLocaleString()} ${stockItem?.reorderLevelUoMName} of ${stockItem?.reorderLevelUoMFactor}` : "",
```
Comment (akileng56) [inline owa/src/features/stockItems/StockItemTable.tsx:151]: Same here

File: owa/src/features/stockItems/StockItemTable.tsx [hunk 12]
```diff
@@ -146,9 +146,9 @@ const StockItemTable: React.FC<StockItemTableProps> = ({
       retired: stockItem?.voided ? t("stockmanagement.yes") : t("stockmanagement.no"),
       isdrug: stockItem?.drugUuid ? t("stockmanagement.yes") : t("stockmanagement.no"),
       preferredVendorName: stockItem?.preferredVendorName,
-      defaultStockOperationsUoMName: stockItem?.defaultStockOperationsUoMName,
+      defaultStockOperationsUoMName: `${stockItem?.defaultStockOperationsUoMName} of ${stockItem?.defaultStockOperationsUoMFactor}`,
```
Comment (akileng56) [inline owa/src/features/stockItems/StockItemTable.tsx:149]: Same here

**Review by akileng56 — CHANGES_REQUESTED** [review]

**Comment by slubwama** [comment]
@akileng56 @Mwanje  resolved all your comments

**Pull request merged**